### ISSUE TITLE
I can't fine Instance.cs

### ISSUE THREAD

**Issue by LitsaM** [xBimTeam/XbimWindowsUI#6]
Hi, I tried to run Xbim.Windows.UI.Nuget and I changed the code so I can view the model without the Menu Items, but I get this error that it can't find the Instance.cs
'e:\src\codeplex_svn\codeplex\EsentInterop\Instance.cs'. 

The project stops in Xbim.IO.IfcPersistedInstanceCache.cs
on line 665 -  var jetInstance = new Instance(instanceName+guid);  

Any Idea how I can find that file or how can I create a new one?

**Comment by andykazmaier** [comment]
Hi,

I think you are talking about running the XbimXplorer project that is part of the Windows.UI nuget package.

Were you able to build, run, and open IFC files prior to making code changes?  When you say you made changes to view the model without the menu items, do you mean that you just commented out some XAML?

That error makes me think that your references are not set up properly, because it is referring to the Esent.Interop reference.  When you expand the references for that project, does it look fine, or are there yellow exclamation points over many or all of the references?

Andy

**Comment by LitsaM** [comment]
Hi Andy thank you for your reply,

Yes I am talking about XbimXplorer in Windows.UI.Nuget,  I was able to build, run and open Ifc files and then I started making changes. I did commented out some XAML and now I am trying to change the "OpenIfcFile" function and call it in  "public XplorerMainWindow()" in "XplorerMainWindow.xaml.cs" without using worker and EventArguments.
I attached the changes that I made, and you can also see some references on the right and there are no yellow exclamation points.

![screenshot 2015-06-30 12 39 24](https://cloud.githubusercontent.com/assets/13109297/8436280/12bff256-1f25-11e5-95e4-516f8a6c1d12.png)

the rest part of code in this function I commented out.

**Comment by andykazmaier** [comment]
When you step through the code, where is it failing?

**Comment by LitsaM** [comment]
Yesterday was here
![screenshot 2015-06-30 12 51 43](https://cloud.githubusercontent.com/assets/13109297/8436509/d249f224-1f26-11e5-8408-affa2662a95e.png)

with this Error
![screenshot 2015-06-30 12 48 42](https://cloud.githubusercontent.com/assets/13109297/8436521/e4683f56-1f26-11e5-972b-aec0b646a7e3.png)

but now I removed Essent.Interop.dll, I added it again, and I Run it again and now I get this
![screenshot 2015-06-30 12 50 25](https://cloud.githubusercontent.com/assets/13109297/8436542/fd498746-1f26-11e5-999b-daea7bd508d6.png)

**Comment by LitsaM** [comment]
But if I add a breakpoint on line 665, and then press "Step Into (F11)" I still get the 1st Error with the Instance.cs

**Comment by andykazmaier** [comment]
Yeah, you're making a lot of changes that I would not do, like not using the worker.  And you've commented out almost all of the exception handling, which is not causing the problem, but it makes me wonder why.

Maybe your filename isn't being generated properly through your concatenation, and then the temp file is bad as well.  But the error that you're seeing is when it is trying to start building the database in the temp file.  I would step through the code where the filename is generated, and the tempfile is generated.  If you have old temp files (which aren't being cleaned up because that code isn't called since the app breaks), that might be a problem as well.

Bottom line is that you're not missing any files.  It's related to the changes that you're making.  Maybe start back from scratch, and slowly walk through your changes again until it breaks.

If you start back from scratch, maybe try hardcoding your filename first and isolate that.  See that work, and then clean up the xaml.  You should be able to do that without changing any of the worker code.

**Comment by LitsaM** [comment]
Thank you Andy I appreciate your help and I will try that now.
 But to make sure, you are saying that if I do all that, I will view the model automatically without buttons and menu Items and by using worker?

**Comment by andykazmaier** [comment]
Not exactly.  What I'm saying is that you should find the place in code where the filename is returned from the open file dialog.  Then just substitute that variable with your hardcoded filename (on the next line, for example).  If you can't get that to work, then you've found your first problem.

Use the open file menu item to start the process, but understand that whatever file you select is irrelevant. Then you can work to make modifications so that code is called automatically, and so that you only see the interface you want to see.

I'm sure there are many ways to debug your problem - this is the approach I would start with personally.

Good luck.

**Comment by LitsaM** [comment]
Ok Thank you so much. I will try that

**Comment by LitsaM** [comment]
Ok so I took your advice and tried to start from scratch, so I removed all the comments and made them as they were on the project that I was trying to change, and then I downloaded a new WindowsUI.Nuget. I made it run and I had both project open and I added a break point at Function "LoadAnyModel" and then I was Stepping into..

Both of the projects were working but mine wouldn't display the model even using the menu items, so I did the stepping work line by line to figured out why. I realized when my steps reached the lines with any Api's and Instances and JetApi, I got the errors that i had mentioned above (that it can't find those files) so I was just pressing "Step Over" and the project continued normally. Then I found out that I had commented " _worker.DoWork += OpenIfcFile;" in function "LoadAnyModel" (i don't remember why, and when I did that) so I uncommented and then my project worked and the model was loaded on start-up without using the menu List....

So problem solved and thank you sooooo much for the help Andy :)

**Comment by andykazmaier** [comment]
Hey, great news!!

Don't forget to close the issue if you're all done with it.  Good luck :)

**Comment by LitsaM** [comment]
Tahnk you :)
and Yes I was trying to close the issue or mark it as done but I can't find how.
Can you help me please? or tell me were can I find the steps coz I searched for it but couldn't find anything

**Comment by andykazmaier** [comment]
Oh, yes.  At the very bottom of the page, next to the comment button, there should be a "Close Issue" button.

**Comment by LitsaM** [comment]
Yea You are right. I haven't seen that coz If you type something it changes to "Close and comment"..
Thank you Andy,
Have a nice day :)
